src/sentry/dynamic_sampling/tasks/boost_low_volume_projects.py

```diff
@@ -398,6 +398,10 @@ def adjust_sample_rates_of_projects(
         )
         return
 
+    # The rebalancing will not work (or would make sense) when we have only projects with zero-counts.
+    if not any(count > 0 for _, count, _, _ in projects_with_tx_count):
+        return
+
     projects_with_counts = {
         project_id: count_per_root for project_id, count_per_root, _, _ in projects_with_tx_count
     }
```

Comment:
Could this be slightly more readable?

```suggestion
    projects_with_counts = {
        project_id: count_per_root for project_id, count_per_root, _, _ in projects_with_tx_count
    } 
    # The rebalancing will not work (or would make sense) when we have only projects with zero-counts.
    if sum(projects_with_counts.values()) == 0:
        return
```